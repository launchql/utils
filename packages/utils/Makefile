EXTENSION = launchql-utils
DATA = sql/launchql-utils--0.3.4.sql

PG_CONFIG = pg_config
PGXS := $(shell $(PG_CONFIG) --pgxs)
include $(PGXS)
  