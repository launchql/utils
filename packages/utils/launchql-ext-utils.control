# launchql-ext-utils extension
comment = 'launchql-ext-utils extension'
default_version = '0.1.17'
module_pathname = '$libdir/launchql-ext-utils'
requires = 'plpgsql'
relocatable = false
superuser = false
  