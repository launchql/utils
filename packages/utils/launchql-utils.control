# launchql-utils extension
comment = 'launchql-utils extension'
default_version = '0.1.29'
module_pathname = '$libdir/launchql-utils'
requires = 'plpgsql'
relocatable = false
superuser = false
  