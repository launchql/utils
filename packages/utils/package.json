{
  "name": "@launchql/utils",
  "version": "0.4.6",
  "description": "utils",
  "author": "Dan Lynch <pyramation@gmail.com>",
  "main": "dist/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/index.d.ts",
  "publishConfig": {
    "access": "public",
    "directory": "dist"
  },
  "scripts": {
    "copy": "copyfiles -f ../../../LICENSE README.md package.json dist",
    "clean": "rimraf dist/**",
    "prepare": "npm run build",
    "build": "npm run clean; tsc; tsc -p tsconfig.esm.json; npm run copy",
    "build:dev": "npm run clean; tsc --declarationMap; tsc -p tsconfig.esm.json; npm run copy",
    "lint": "eslint . --fix",
    "test": "jest",
    "test:watch": "jest --watch"
  },
  "devDependencies": {
    "@types/jest": "^29.5.11",
    "@types/node": "^20.12.7",
    "@launchql/cli": "1.3.5",
    "copyfiles": "^2.4.1",
    "eslint": "^8.56.0",
    "eslint-config-prettier": "^9.1.0",
    "jest": "^29.6.2",
    "prettier": "^3.0.2",
    "rimraf": "4.4.1",
    "ts-jest": "^29.1.1"
  },
  "dependencies": {
    "pgsql-test": "^2.1.14"
  },
  "gitHead": "bc0f36b82f3e103a79c3e75adae8fd0434f1bdb1"
}
