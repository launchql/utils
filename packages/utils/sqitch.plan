%syntax-version=1.0.0
%project=launchql-utils
%uri=launchql-utils
  
schemas/utils/schema 2017-08-11T08:11:51Z skitch <skitch@5b0c196eeb62> # add schemas/utils/schema
schemas/utils/procedures/mask_pad [schemas/utils/schema] 2017-08-11T08:11:51Z skitch <skitch@5b0c196eeb62> # add schemas/utils/procedures/mask_pad