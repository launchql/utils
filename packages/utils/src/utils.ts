
/**
 * Utility functions that mirror the SQL functions in the utils schema
 */
export const utils = {
};
