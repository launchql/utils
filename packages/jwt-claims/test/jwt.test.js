import { getConnections } from './utils';

let db, teardown;
const apis = {};
const jwt = {
  user_id: 'b9d22af1-62c7-43a5-b8c4-50630bbd4962',
  database_id: '44744c94-93cf-425a-b524-ce6f1466e327',
  group_ids: [
    'f12c75c2-47d5-43fd-9223-d42d08f51942',
    'd96d32b4-e819-4cb1-8a27-e27e763e0d7f',
    'c8a27b31-1d40-4f40-9cb0-e96a44e68072'
  ]
};
beforeAll(async () => {
  ({ db, teardown } = await getConnections());
  apis.public = db.helper('jwt_public');
  apis.private = db.helper('jwt_private');
});

afterAll(async () => {
  try {
    //try catch here allows us to see the sql parsing issues!
    await teardown();
  } catch (e) {
    // noop
  }
});

beforeEach(async () => {
  await db.beforeEach();
});

afterEach(async () => {
  await db.afterEach();
});

it('get values', async () => {
  db.setContext({
    'jwt.claims.user_agent':
      'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36',
    'jwt.claims.ip_address': '127.0.0.1',
    'jwt.claims.database_id': jwt.database_id,
    'jwt.claims.user_id': jwt.user_id,
    'jwt.claims.group_ids': `{${jwt.group_ids.join(',')}}`
  });
  const user_agent = await apis.public.call('current_user_agent');
  const ip_address = await apis.public.call('current_ip_address');
  const database_id = await apis.private.call('current_database_id');
  const group_ids = await apis.public.call('current_group_ids');
  const user_id = await apis.public.call('current_user_id');
  expect({ user_agent }).toMatchSnapshot();
  expect({ ip_address }).toMatchSnapshot();
  expect({ database_id }).toMatchSnapshot();
  expect({ group_ids }).toMatchSnapshot();
  expect({ user_id }).toMatchSnapshot();
});
