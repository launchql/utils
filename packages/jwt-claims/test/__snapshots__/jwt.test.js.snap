// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get values 1`] = `
Object {
  "user_agent": Array [
    Object {
      "current_user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36",
    },
  ],
}
`;

exports[`get values 2`] = `
Object {
  "ip_address": Array [
    Object {
      "current_ip_address": "127.0.0.1",
    },
  ],
}
`;

exports[`get values 3`] = `
Object {
  "database_id": Array [
    Object {
      "current_database_id": null,
    },
  ],
}
`;

exports[`get values 4`] = `
Object {
  "user_id": Array [
    Object {
      "current_user_id": null,
    },
  ],
}
`;
